logprep/connector/s3/output.py

```diff
@@ -195,52 +198,57 @@ def _write_to_s3_resource(self, document: dict, prefix: str):
         ----------
         document : dict
            Document to store.
-
-        Returns
-        -------
-        Returns True to inform the pipeline to call the batch_finished_callback method in the
-        configured input
         """
         prefix = self._add_dates(prefix)
         prefix = f"{self._base_prefix}{prefix}"
         self._message_backlog[prefix].append(document)
 
-        backlog_count = self._current_backlog_count + 1
-        if backlog_count == self._config.message_backlog_size:
-            if self._writing_thread is None or not self._writing_thread.is_alive():
-                message_backlog = deepcopy(self._message_backlog)
-                self._writing_thread = threading.Thread(
-                    target=self._write_document_batches, args=(message_backlog,)
-                )
-                self._writing_thread.start()
-                return True
-        self._current_backlog_count = backlog_count
-        return False
-
-    def _write_document_batches(self, message_backlog):
-        self._logger.info(f"Writing {self._current_backlog_count + 1} documents to s3")
-        for prefix_mb, document_batch in message_backlog.items():
+        if self._get_backlog_size(self._message_backlog) >= self._config.message_backlog_size:
+            self._write_backlog()
+
+    @staticmethod
+    def _get_backlog_size(message_backlog: dict):
+        return sum(len(values) for values in message_backlog.values())
+
+    def _write_backlog(self):
+        """Write to s3 if it is not already writing."""
+        if not self._message_backlog:
+            return
+
+        self._bulk()
+
+    def _bulk(self):
+        self._logger.info(
+            "Writing %s documents to s3", self._get_backlog_size(self._message_backlog)
+        )
```

Comment:
is this really necessary? Or schould this be a DEBUG info?

Reply:
It is helpful to see the exact timestamps when something was written and which warnings or errors happened in relation to that.
Our debug output is very verbose at the moment, so that it's hard to use it for things like this.